packages/composables/src/useProductPrice.ts

```diff
@@ -71,37 +71,49 @@ export function useProductPrice(
     Schemas["CalculatedPrice"]["referencePrice"] | undefined
   > = computed(() => _real?.value?.referencePrice);
 
-  const _displayParent: ComputedRef<boolean> = computed(
-    () =>
-      !!product.value?.variantListingConfig?.displayParent &&
-      product.value?.parentId === null,
-  );
+  // const _displayParent: ComputedRef<boolean> = computed(
+  //   () =>
+  //     !!product.value?.variantListingConfig?.displayParent &&
+  //     product.value?.parentId === null,
+  // );
 
-  const displayFrom: ComputedRef<boolean> = computed(
-    () =>
-      (product.value?.calculatedPrices?.length ?? 0) > 1 ||
-      !!(_displayParent.value && displayFromVariants.value),
-  );
+  const displayFrom: ComputedRef<boolean> = computed(() => {
+    // Change me mind:
+    //  `!!(_displayParent.value && displayFromVariants.value)` will never return true
+    // - _displayParent.value expects `product.value?.parentId === null`
+    // - displayFromVariants.value expects `!!product.value?.parentId`
+
+    // return (
+    //   (product.value?.calculatedPrices?.length ?? 0) > 1 ||
+    //   !!(_displayParent.value && displayFromVariants.value)
+    // );
+
+    return (product.value?.calculatedPrices?.length ?? 0) > 1;
+  });
 
   const displayFromVariants: ComputedRef<number | false | undefined> = computed(
-    () =>
-      !!product.value?.parentId &&
-      product.value?.calculatedCheapestPrice?.hasRange &&
-      _real?.value?.unitPrice !== _cheapest?.value?.unitPrice &&
-      _cheapest?.value?.unitPrice,
+    () => {
+      return (
+        !!product.value?.parentId &&
+        product.value?.calculatedCheapestPrice?.hasRange &&
+        _real?.value?.unitPrice !== _cheapest?.value?.unitPrice &&
+        _cheapest?.value?.unitPrice
+      );
+    },
   );
 
   const _price: ComputedRef<Schemas["CalculatedPrice"] | undefined> = computed(
     () => {
       if (displayFrom.value && getProductTierPrices(product.value).length > 1) {
-        const lowest = product.value?.calculatedPrices?.reduce(
-          (previous, current) => {
-            return current.unitPrice < previous.unitPrice ? current : previous;
-          },
-        );
-        return (
-          lowest || (_cheapest.value as unknown as Schemas["CalculatedPrice"])
-        );
+        return product.value?.calculatedPrices?.reduce((previous, current) => {
+          return current.unitPrice < previous.unitPrice ? current : previous;
+        });
+
+        // Change my mind:
+        // There is no possibility to return _cheapest.value because of the getProductTierPrices that is iterating over calculatedPrices
```

Comment:
If somehow in the `calculatedPrices` the `unitPrice` would be `0` then in theory it is possible to return `_cheapest`. If that is the case anywhere? I'm not sure. I'd leave this logic (without casting to unknown and CalculatedPrice) and create a follow-up issue for the discovery of that case

Reply:
now, when the types reflect the actual shape of the data from API we can trust it, so removing redundant code can be done. I've checked the preview and compared the variants, from prices and so on. The prices are displayed correctly - in comparison to `main` branch which reflects the storefront behavior.